login/service.go

```diff
@@ -337,6 +339,166 @@ func (keycloak *KeycloakOAuthProvider) Perform(ctx *app.LoginLoginContext, confi
 	return ctx.TemporaryRedirect()
 }
 
+// PerformAuthorize takes care of authorize action
+func (keycloak *KeycloakOAuthProvider) PerformAuthorize(ctx *app.AuthorizeAuthorizeContext, config oauth.OauthConfig, serviceConfig LoginServiceConfiguration) error {
+	/* Compute all the configuration urls */
+
+	validRedirectURL := serviceConfig.GetValidRedirectURLs()
+
+	state := ctx.Params.Get("state")
+	code := ctx.Params.Get("code")
+
+	log.Debug(ctx, map[string]interface{}{
+		"code":  code,
+		"state": state,
+	}, "login request received")
+
+	if code != "" {
+		// After redirect from oauth provider
+		log.Debug(ctx, map[string]interface{}{
+			"code":  code,
+			"state": state,
+		}, "redirected from oauth provider")
+
+		authCode := &app.AuthorizationCode{
+			Code:  *ctx.Code,
+			State: *ctx.State,
+		}
+		return ctx.OK(authCode)
```

Comment:
We should redirect to the original redirect URL and pass the code there instead of returning 200.

Reply:
https://github.com/fabric8-services/fabric8-auth/pull/210/commits/6597103ce44a1b789b4de9f9c257a3ae1532b596